### PR TITLE
Stop all next tick

### PR DESCRIPTION
**Pull request by mzgoddard**
### Resolves

Part of https://github.com/LLK/scratch-vm/issues/1478

### Proposed Changes

Match Scratch 2.0 `Interpreter.stopAllThreads()` behaviour.

### Reason for Changes

In Scratch 2.0, `stop all` only immediately stops the active thread and removes the remaining threads from executing next tick. The remaining threads are able to execute one last time. It is also possible because of this that the remaining threads can create new threads that will in fact execute next tick.

### Test Coverage

Add a execute integration test that makes sure threads after stop all get one more execution and that they can actually create new threads that will execute in a later tick.

The behaviour is demonstrated in Scratch 2.0 project 13701368.

### CODE REVIEW

File: src/engine/runtime.js [hunk 1]
```diff
@@ -1494,9 +1494,14 @@ class Runtime extends EventEmitter {
                 newTargets.push(this.targets[i]);
             }
         }
+
```
Comment (kchadha) [inline src/engine/runtime.js]: stray newline

**Comment by BryceLTaylor** [comment]
in Scratch 2.0 if in the second thread you call another broadcast and the second broadcast has a forever in it, stop all doesn't stop that other forever:
https://scratch.mit.edu/projects/260463874/
endthreadtest
The above changes don't cause that to be true.
The cat does the full Meow in this example.

Here that situation is in 3.0 after this PR:
endthreadtestin3

The cat meows fully but does not enter the change costume loop.

Also, in 2.0, if you use a custom procedure to create the other thread and the custom procedure has a broadcast, the when I receive does not run.
https://scratch.mit.edu/projects/260471599/#editor
endthreadtest2

In that gif the cat meows briefly and then is stopped.

Here's that situation in 3.0 after this PR:
endthreadtest2in3

This behaves the same in 3.0 as 2.0 with half a meow.

**Comment by kchadha** [comment]
@mzgoddard, marking this as needs work because of the issue that @BryceLTaylor mentioned with the   custom procedures. I would be open to filing that as a separate issue and landing this one if the fix is particularly complicated.

**Comment by ktbee** [comment]
I think the issues @BryceLTaylor noted from when @mzgoddard first submitted this PR have been fixed! When I download this branch and merge in the latest changes from `develop`, I find that the example projects that @BryceLTaylor provided have compatible behavior with Scratch 2, ie the hat block stack now fires in #260463874 despite the "stop all" block. 

I believe [our recent fix to start hat threads in `startHats`](https://github.com/LLK/scratch-vm/pull/1943) has made this PR's behavior more compatible with Scratch 2. Now those threads are started before we have a chance to remove them from the runtime's threads array in `stopAll`.

@BryceLTaylor are these examples now resolved? Or am I missing some incompatible behavior? 

/cc @kchadha

**Comment by mzgoddard** [comment]
- Fix conflict (It was an added comment.)

**Comment by BryceLTaylor** [comment]
@mzgoddard  @kchadha I think this looks pretty good!

**Review by kchadha — APPROVED** [review]
LGTM! Thanks @mzgoddard!

**Pull request merged**